keyboards/keycapsss/kimiko/keymaps/rgb_matrix/keymap.c

```diff
@@ -0,0 +1,473 @@
+}
+
+void render_status_main(void) {
+    render_space();
+    render_space();
+    render_logo();
+    render_space();
+    render_layer_state();
+    render_space();
+    render_mod_status_gui_alt(get_mods()|get_oneshot_mods());
+    render_mod_status_ctrl_shift(get_mods()|get_oneshot_mods());
+}
+
+void render_status_secondary(void) {
+    render_space();
+    render_space();
+    render_logo();
+    render_space();
+}
+
+bool oled_task_user(void) {
+    if (is_keyboard_master()) {
+        render_status_main();  // Renders the current keyboard state (layer, lock, caps, scroll, etc)
+    } else {
+        render_status_secondary();
+    }
+    return false;
+}
+
+#endif
+
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    // Encoder on master side
+    if (index == 0) {
+        switch (get_highest_layer(layer_state)) {
+            // If the Default (QWERTY) layer is active
+            case _QWERTY:
+                // Arrow Up/Down
+                if (clockwise) {
+                    tap_code(KC_DOWN);
+                } else {
+                    tap_code(KC_UP);
+                }
+                break;
+
+            // If the RAISE layer is active
+            case _RAISE:
+                // Switch browser tabs
+                if (clockwise) {
+                    tap_code16(LCTL(KC_TAB));
+                } else {
+                    tap_code16(RCS(KC_TAB));
+                }
+                break;
+            // If the ADJUST layer is active
+            case _ADJUST:
+                // RGB brightness up/down
+                if (clockwise) {
+                    rgblight_decrease_val(); // tap_code(RGB_VAD);
+                } else {
+                    rgblight_increase_val(); // tap_code(RGB_VAI);
+                }
+                break;
+        }
+    }
+    // Encoder on slave side
+    else if (index == 1) {
+        switch (get_highest_layer(layer_state)) {
+            // If the Default (QWERTY) layer is active
+            case _QWERTY:
+                // Scroll by Word
+                if (clockwise) {
+                    tap_code16(LCTL(KC_RGHT));
+                } else {
+                    tap_code16(LCTL(KC_LEFT));
+                }
+                break;
+
+            // If the LOWER layer is active
+            case _LOWER:
+                // Volume up/down
+                if (clockwise) {
+                    tap_code(KC_VOLU);
+                } else {
+                    tap_code(KC_VOLD);
+                }
+                break;
+
+            // If the ADJUST layer is active
+            case _ADJUST:
+                // RGB hue up/down
+                if (clockwise) {
+                    // tap_code(RGB_HUI);
+                    rgblight_increase_hue();
+                } else {
+                    // tap_code(RGB_HUD);
+                    rgblight_decrease_hue();
+                }
+                break;
+        }
+    }
+    return true;
```

Comment:
If the keyboard has a default behavior, setting this to true will trigger both functions. 
```suggestion
    return false;
```

Also, if you wait, "encoder map" lands at the end of the month, and that may be much closer to what you want here.